lib/handler/pool.ex

```diff
@@ -65,7 +65,13 @@ defmodule Handler.Pool do
   @spec async(pool(), (() -> any()), opts()) :: {:ok, reference()} | {:reject, exception}
   def async(pool, fun, opts) do
     validate_pool_opts!(opts)
-    GenServer.call(pool, {:run, fun, opts}, 1_000)
+
+    try do
+      GenServer.call(pool, {:run, fun, opts}, 1_000)
+    catch
+      :exit, {:noproc, _} ->
+        {:reject, NoWorkersAvailable.exception(message: "No Pools Available")}
```

Comment:
Me, marcello and Jonner chatted this through and we want to improve how we accumulate/return error messages in the async call, but it falls outside the scope of this PR and isn't a blocker for now